tests/run-make/print-target-cpus-native/rmake.rs

```diff
@@ -0,0 +1,37 @@
+//@ ignore-cross-compile
+
+use run_make_support::{assert_contains_regex, rfs, rustc, target};
+
+// Test that when querying `--print=target-cpus` for a target with the same
+// architecture as the host, the first CPU is "native" with a suitable remark.
+
+fn main() {
+    let expected = r"^Available CPUs for this target:
+    native +- Select the CPU of the current host \(currently [^ )]+\)\.
+";
+
+    // Without an explicit target.
+    rustc().print("target-cpus").run().assert_stdout_contains_regex(expected);
+
+    // With an explicit target that happens to be the host.
+    let host = target(); // Because of ignore-cross-compile, assume host == target.
+    rustc().print("target-cpus").target(host).run().assert_stdout_contains_regex(expected);
+
+    // With an explicit output path.
+    rustc().print("target-cpus=./xyzzy.txt").run().assert_stdout_equals("");
+    assert_contains_regex(rfs::read_to_string("./xyzzy.txt"), expected);
+
+    // Now try some cross-target queries with the same arch as the host.
+    // (Specify multiple targets so that at least one of them is not the host.)
+    let cross_targets: &[&str] = if cfg!(target_arch = "aarch64") {
+        &["aarch64-unknown-linux-gnu", "aarch64-apple-darwin"]
```

Comment:
Question: ... would this need `needs-llvm-component`

Reply:
Hmm, I'm not sure.

My guess is no, because it's guaranteed by `ignore-cross-compile` that we at least have the relevant arch component for the host system, and we're only testing other targets with the same arch as the host.

But I know very little about LLVM components.

Reply:
I asked because I have dealt with run-make tests that seemed innocent (like see https://github.com/rust-lang/rust/issues/129390)

Reply:
If the condition will usually be true in CI and locally, then I don't mind adding the directive as a precaution.

Reply:
Yeah, it's mostly just case some contributors do build local llvms without unneeded components.